src/scripts/core/models/grid.ts

```diff
@@ -36,41 +32,73 @@ export class Grid extends Container {
 
     }
 
-    private clickCheck(tokenX: number, tokenY: number): void {
+    private clickCheck(targetToken: Token): void {
         //on first click, despite being defined in the constructor, activeToken appears undefined
         if(this.activeToken === undefined) {
             this.activeToken = 0;
         }
 
-        const location = [tokenX, tokenY];
-        this.matchCheck(location);
-
-        // if(this.activeToken === 0) {
-        //     this.activeToken = 1; 
-        //     eventEmitter.emit('tokenFirstClicked', tokenX, tokenY);
-        //     console.log("FirstClick", this.activeToken)
-        //     return;
-        // }
-
-        // if(this.activeToken === 1) {
-        //     this.activeToken = 2; 
-        //     eventEmitter.emit('tokenSecondClicked', tokenX, tokenY); 
-        //     console.log("SecondClick", this.activeToken)
-        //     return; 
-        // }
-    }
+        //on FirstClick
+        if((this.activeToken === 0) && (this.firstTokenLocation[0] === -1) && (this.firstTokenLocation[0] === -1)) {
+            targetToken.highLight();
+            this.activeToken = 1;
+            this.firstTokenLocation = targetToken.getLocation();
+            return;
+        }
+
+        //on SecondClick
+        if(this.activeToken === 1) {
+            targetToken.highLight();
+            this.activeToken = 2;
+
+            //Identify first Token
+            const firstX = this.firstTokenLocation[0];
+            const firstY = this.firstTokenLocation[1];
+            const firstTokenCopy = this.getToken(firstX, firstY);
+            const firstSkIndex = firstTokenCopy.getSkIndex();
+
+            //Identify second Token
+            const secondX = targetToken.getLocation()[0];
+            const secondY = targetToken.getLocation()[1];
+            const secondSkIndex = this.getToken(secondX, secondY).getSkIndex();
 
-    private swapPrepare(): void {
+            // //Swap Tokens
+            const tempFirstToken = this.getToken(firstX, firstY);
+            const tempSecondToken = this.getToken(secondX, secondY);
+            
+
+            //Swapping isn't working as expected
+            let targetFirst = this.getToken(firstX, firstY);
+            targetFirst.setToken(secondSkIndex);
+            let targetSecond = this.getToken(secondX, secondY);
+            targetSecond.setToken(firstSkIndex);
+
+            this.matchCheck(this.getToken(firstX, firstY));
+            this.matchCheck(this.getToken(secondX, secondY));
+
+            this.nukeBoard();
+
+            this.firstTokenLocation = [-1, -1];
+            this.activeToken = 0;
+            return;
+        }
```

Comment:
Definitely some cleanup that can be done in this whole section.

For example you call `this.getToken(firstX, firstY)` 4 times in here and assign that same result into 3 different variables.

I think Mollets point above stands true.
You are receiving the clicked Token in this event so it may be cleaner to just:
- on first click, store the first token in a variable.
- on second click you already have the second token passed through so can read the first token from your variable and never need to call `getToken` at all.
- just make a copy of each tokens skin, then re-assign each token with the other skin.